### PR TITLE
Fixed trailing zero bug in CalculatorImpl [Issue #77]

### PR DESCRIPTION
**Pull request by jusleg**
fixes: #77 

### WHAT kind of change does this PR introduce?
Added a new method `doubleToStringWithGivenDigits` to the formatter to add the proper trailing digits. It is used when setting the value after the user pressed a digit. I added a UI test in `MainActivityTest` to cover this case and renamed the neighboring tests so that they contained the word "test" in the method name (proper test conventions). I also added test to cover all the functionalities of the formatter. The formatter isn't legacy code anymore 🎉 

### HOW is this accomplished?
  * new method
  * new test
  * new test class for formatter (inside commons)

### Checklist
- [x] Docs have been added / updated to reflect the changes
- [x] I have reviewed and tested the changes :white_check_mark:
- [x] I have squashed my commits to have a reasonable amount of commits
- [x] All of the code I have written adhere to the coding conventions outlined in the coding conventions section of the wiki

### CODE REVIEW

File: app/src/androidTest/java/com/simplemobiletools/calculator/MainActivityTest.java [hunk 1]
```diff
@@ -36,12 +36,30 @@ public void addDigits() {
...
     @Test
-    public void removeLeadingZero() {
+    public void removeLeadingZeroTest() {
         press(R.id.btn_0);
         press(R.id.btn_5);
         checkResult("5");
     }
 
+    @Test
+    public void removeTrailingZero() {
```
Comment (almiche) [inline app/src/androidTest/java/com/simplemobiletools/calculator/MainActivityTest.java]: Thorough!

File: commons/src/main/java/com/simpletools/calculator/commons/helpers/CalculatorImpl.kt [hunk 2]
```diff
@@ -150,4 +151,8 @@ class CalculatorImpl(calculator: Calculator) {
         firstNumber *= -1
         setValue(Formatter.doubleToString(firstNumber))
     }
+
+    private fun formatUserInputTrailingZero(number: Double): String{
```
Comment (almiche) [inline commons/src/main/java/com/simpletools/calculator/commons/helpers/CalculatorImpl.kt]: Why don't we include this in the formatter

**Comment by almiche** [comment]
✅| Styling is adequate and follows coding conventions outlined by the [Kotlin Foundation](https://kotlinlang.org/docs/reference/coding-conventions.html)
  Everything conforms to the guidelines

✅| New methods and functionalities have appropriate tests (testing edge cases + good coverage)
  A test was indeed introduced for the feature

✅| Branch has an acceptable amount of commits with descriptive names and

✅| PR description matches the code outlined a has mentions to issues

✅| This PR attempts to implement a specific feature (no code blob or epic pr)

Status: **Changes needed**
Comments: Waiting on responses about my review

File: commons/src/main/java/com/simpletools/calculator/commons/helpers/CalculatorImpl.kt (hunk 2 above)
Comment (0matleb2) [reply commons/src/main/java/com/simpletools/calculator/commons/helpers/CalculatorImpl.kt]: Agreed, this function would be appropriately encapsulated in the Formatter with a name like doubleFixedPointToString.

Comment (jusleg) [reply commons/src/main/java/com/simpletools/calculator/commons/helpers/CalculatorImpl.kt]: Fixed plus I added test for every formatter method

File: app/src/androidTest/java/com/simplemobiletools/calculator/MainActivityTest.java [hunk 3]
```diff
@@ -21,7 +21,7 @@
     @Rule public final ActivityTestRule<MainActivity> activity = new ActivityTestRule<>(MainActivity.class);
 
     @Test
-    public void addDigits() {
+    public void addDigitsTest() {
```
Comment (0matleb2) [inline app/src/androidTest/java/com/simplemobiletools/calculator/MainActivityTest.java]: *Style comment*. The current naming convention for tests are descriptive instead of prescriptive, ie. this is an `addDigitsTest()`, instead of, do `testAddDigits()`. Furthermore, the names could also include the expected result, ie. `testAddDigitsShouldDisplayDigits()`. Thoughts on changing this convention?

Comment (almiche) [reply app/src/androidTest/java/com/simplemobiletools/calculator/MainActivityTest.java]: You raise a valid point should be something to look into further on.
